fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/ClientTest.java

```diff
@@ -143,4 +143,36 @@ public void testClientAsEntityStatus() {
 
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testPendingOnlyClientRequest() {
+
+        Random rand = new Random();
+        // Add a few clients to the server and activate a random amount of them
+        for (int i = 0; i < 15; i++) {
+            final Integer clientId = ClientHelper.createClientAsEntity(this.requestSpec, this.responseSpec);
+            if (rand.nextInt(10) % 2 == 0) {
+                // Takes Client to pending status
+                this.clientHelper.closeClient(clientId);
+                this.clientHelper.reactivateClient(clientId);
+            }
+            // Other clients stay in Active status
+        }
+        List<HashMap<String, Object>> clientsRecieved = (List<HashMap<String, Object>>) clientHelper.getClientWithStatus(50, "pending");
+
```

Comment:
Should we not check that we receive at least some clients? As far as I can see, this test will still pass if the search doesn't actually work and returns nothing... 

Ideally we should check that we receive some of the same clients that we set to status pending above - but at the very least we should check that we've received something...